spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1806,13 +1843,19 @@ public void handleConsumeOk(String consumerTag) {
 				latch.countDown();
 			}
 
+			@Override
+			public void handleDelivery(String consumerTag, Envelope envelope, BasicProperties properties, byte[] body)
+					throws IOException {
+				future.complete(new Delivery(envelope, properties, body));
+			}
+
 		};
 		channel.basicConsume(queueName, consumer);
 		if (!latch.await(10, TimeUnit.SECONDS)) {
```

Comment:
We're blocking downstream anyway on the future. This makes life a bit easier because we don't need to deal with races.